benchmark/benchmark.go

```diff
@@ -232,40 +233,60 @@ func DoUnaryCall(tc testgrpc.BenchmarkServiceClient, reqSize, respSize int) erro
 }
 
 // DoStreamingRoundTrip performs a round trip for a single streaming rpc.
-func DoStreamingRoundTrip(stream testgrpc.BenchmarkService_StreamingCallClient, reqSize, respSize int) error {
+func DoStreamingRoundTrip(stream testgrpc.BenchmarkService_StreamingCallClient, reqSize, respSize, streamCount int) error {
```

Comment:
Sorry, I think I missed this in my previous iteration. The way this is implemented currently, it creates a _single_ stream and performs multiple sends and receives on the _same_ stream. What I believe you are trying to add is to actually create multiple streams. In that case, we would have to change the signature of `DoStreamingRoundTrip()` to accept a `grpc.ClientConnInterface` instead of a `testgrpc.BenchmarkService_StreamingCallClient` and create `streamCount` number of streams inside this function (from goroutines), and perform sends and receives on them.

Also, the `Send()` and `Receive()` can be inside the same goroutine instead of the latter being outside the goroutine.

Reply:
> Sorry, I think I missed this in my previous iteration. The way this is implemented currently, it creates a _single_ stream and performs multiple sends and receives on the _same_ stream. What I believe you are trying to add is to actually create multiple streams. In that case, we would have to change the signature of `DoStreamingRoundTrip()` to accept a `grpc.ClientConnInterface` instead of a `testgrpc.BenchmarkService_StreamingCallClient` and create `streamCount` number of streams inside this function (from goroutines), and perform sends and receives on them.
> 
> Also, the `Send()` and `Receive()` can be inside the same goroutine instead of the latter being outside the goroutine.

I would like to create a single stream and perform multiple sends and receives. The current implementation is sufficient for my needs.

Reply:
We have three modes of operation for the benchmark client, controlled by the `-workloads` flag, and in each of these modes, `-maxConcurrentCalls` number of streams are created upfront
- Unary: One goroutine per stream is created, and one RPC is done at a time
- Streaming: One goroutine per stream is created, and one `Send()` and `Recv()` is done at a time
- Unconstrained: Two goroutines per stream are created, one for sending and one for receiving, and two goroutines send and recv independent of the other

Is the unconstrained mode any different from what you are trying to do here?

Reply:
> Is the unconstrained mode any different from what you are trying to do here?

It's almost same. the unconstrained mode can accomplish what I am attempting to do here.